spec/build-npm_spec.sh

```diff
@@ -0,0 +1,217 @@
+      export GITHUB_OUTPUT=/dev/null
+      When call set_npm_version_with_build_id "42"
+      The status should be success
+      The output should include "Replacing version 1.2.3-SNAPSHOT with 1.2.3-42"
+    End
+
+    It 'validates version format correctly'
+      When call check_version_format "1.2.3.4"
+      The status should be success
+      The output should equal ""
+    End
+
+    It 'warns about invalid version format'
+      When call check_version_format "1.2.3-42"
+      The status should be success
+      The stderr should include "WARN: Version '1.2.3-42' does not match the expected format"
+    End
+  End
+
+  Describe 'Build pipeline functions'
+    BeforeEach 'source build-npm/build.sh'
+
+    It 'runs npm install correctly'
+      When call run_npm_install
+      The status should be success
+      The output should include "Installing npm dependencies..."
+      The output should include "npm ci completed"
+    End
+
+    It 'runs npm tests when not skipped'
+      export SKIP_TESTS="false"
+      When call run_npm_tests
+      The status should be success
+      The output should include "Running tests..."
+      The output should include "npm test completed"
+    End
+
+    It 'skips npm tests when requested'
+      export SKIP_TESTS="true"
+      When call run_npm_tests
+      The status should be success
+      The output should include "Skipping tests (SKIP_TESTS=true)"
+    End
+
+    It 'runs npm build correctly'
+      When call run_npm_build
+      The status should be success
+      The output should include "Building project..."
+      The output should include "npm run build completed"
+    End
+
+    It 'runs standard pipeline with sonar and deploy'
+      export SKIP_TESTS="false"
+      export PROJECT="test-project"
+      export BUILD_NUMBER="42"
+      When call run_standard_pipeline true true -Dsonar.projectVersion="1.0.0"
+      The status should be success
+      The output should include "Installing npm dependencies..."
+      The output should include "Running tests..."
+      The output should include "SonarQube scanner completed"
+      The output should include "Building project..."
+    End
+
+    It 'runs standard pipeline without sonar'
+      export SKIP_TESTS="false"
+      When call run_standard_pipeline false false
+      The status should be success
+      The output should include "Installing npm dependencies..."
+      The output should include "Running tests..."
+      The output should include "Building project..."
+      The output should not include "SonarQube scanner"
+    End
+  End
+
+  Describe 'JFrog deployment'
+    BeforeEach 'source build-npm/build.sh'
+
+    It 'fails when missing environment variables'
+      unset ARTIFACTORY_URL
+      unset ARTIFACTORY_DEPLOY_ACCESS_TOKEN
+      When run jfrog_npm_publish
+      The status should be failure
+      The stderr should include "ERROR: Deployment requires ARTIFACTORY_URL and ARTIFACTORY_DEPLOY_ACCESS_TOKEN"
```

Comment:
No, we don't want to test the presence of the variables.
It's redundant, not adding coverage and making the spec file harder to maintain.

The variables are covered by Bash validation at the beginning of the Shell script. That's enough.